.pipelines/templates/soak-test.yml

```diff
@@ -70,10 +56,15 @@ jobs:
             export IDENTITY_VALIDATOR_VERSION="${IMAGE_VERSION}"
             make e2e
           env:
-            SUBSCRIPTION_ID: $(SUBSCRIPTION_ID)
-            AZURE_CLIENT_ID: $(AZURE_CLIENT_ID)
-            AZURE_CLIENT_SECRET: $(AZURE_CLIENT_SECRET)
-            AZURE_TENANT_ID: $(AZURE_TENANT_ID)
+            # used by one of the test cases
             SERVICE_PRINCIPAL_CLIENT_ID: $(SERVICE_PRINCIPAL_CLIENT_ID)
             SERVICE_PRINCIPAL_CLIENT_SECRET: $(SERVICE_PRINCIPAL_CLIENT_SECRET)
           displayName: "Run E2E tests"
+
+        - script: |
+            NAMESPACES="$(kubectl get namespace -ojson | jq -r '.items[].metadata.name' | grep -vE 'kube-.*|default')"
+            if [[ -n "${NAMESPACES:-}" ]]; then
+              kubectl delete namespace $(echo NAMESPACES | tr '\n' ' ')
+            fi
+          displayName: "Remove dangling namespaces"
```

Comment:
Are there dangling namespaces today? If there are we should ensure it's cleaned up in the e2e tests instead of doing it as part of the pipelines

Reply:
I added this to help me automatically remove any dangling namespaces in case of any failure while working on this PR. Let me remove this.